File: Barcos/barcos.cpp
//Jorge Castrillo Rojas - B41570
//Reposición - Herencias

#include <iostream>
using namespace std;
 
class Nacion
{
public:
    Nacion(string pais) //constructor
    {
        cout << "País: " << pais << endl;
    }
};
 
class Tipo: public Nacion //hereda de Nacion
{
public:
    Tipo(string pais, string tipo)
    : Nacion(pais) //constructor
    {
        cout << "Tipo: " << tipo << endl;
    }
};
 
class Clase: public Tipo	//hereda de Tipo
{
public:
    Clase(string pais, string tipo, string clase)
    : Tipo(pais, tipo) //constructor
    {
        cout << "Clase: " << clase << endl;
    }
};

class Barco: public Clase	//hereda de Clase
{
public:
    Barco(string pais, string tipo, string clase, string barco, int year, string destino)
    : Clase(pais, tipo, clase) //constructor
    {
        cout << "Barco: " << barco << endl;
        cout << "Asignado: " << year << endl;
        cout << "Destino: " << destino << endl;
    }
};

//Este main es para mí. Es la primera vez que trabajo con herencias, entonces es para ver si es correcto 
int main()
{
    Barco Formidable("Reino Unido", "Portaaviones", "Illustrious", "Formidable", 1940, "Desguazado, 1953");
}

File: Barcos/barcos.py
##Jorge Castrillo Rojas - B41570
#3Reposición - Herencias

class Nacion():       
    #Constructor 
    def __init__(self, pais): 
        self.pais = pais 
    #
    def getNacion(self):
        print("País:", self.pais)
      
class Tipo(Nacion): 
    #Constructor 
    def __init__(self, pais, tipo):
        Nacion.__init__(self, pais)
        self.tipo = tipo

    def getTipo(self): 
        print("Tipo:", self.tipo)

class Clase(Tipo): 
    #Constructor 
    def __init__(self, pais, tipo, clase):
        Tipo.__init__(self, pais, tipo)
        self.clase = clase

    def getClase(self): 
        print("Clase:", self.clase)

class Barco(Clase): 
    #Constructor 
    def __init__(self, pais, tipo, clase, barco, year, destino):
        Clase.__init__(self, pais, tipo, clase)
        self.barco = barco
        self.year = year
        self.destino = destino

    def getBarco(self): 
        print("Barco:", self.barco)

    def getYear(self): 
        print("Asignado:", self.year)

    def getDestino(self): 
        print("Destino:", self.destino)

   
# Ejemplo para ver el funcionamiento del código 
formidable = Barco("Reino Unido", "Portaaviones", "Illustrious", "Formidable", 1940, "Desguazado, 1953")
formidable.getNacion()
formidable.getTipo()
formidable.getClase()
formidable.getBarco()
formidable.getYear()
formidable.getDestino()
